client/components/calendar/calendar.jsx

```diff
@@ -0,0 +1,116 @@
+import React, {Component} from 'react';
+import Styles from './styles.js';
+import moment from 'moment';
+
+import Week from './week.jsx';
+
+class Calendar extends Component {
+  constructor(props){
+    super(props);
+
+    this.state = {
+      date: moment(),
+      weeks: [],
+      month: '',
+      year: moment().month(moment().month()).format('YYYY'),
+      incrementMonth: 0
+    }
+
+    this.dateRight = this.dateRight.bind(this);
+    this.dateLeft = this.dateLeft.bind(this);
+    this.renderWeeks = this.renderWeeks.bind(this);
+  }
+
+  componentDidMount(){
+    this.renderWeeks(moment().month());
+  }
+
+  async dateRight(){
+    await this.setState({
+      incrementMonth: this.state.incrementMonth + 1
```

Comment:
Use the callback form of `this.setState`: https://reactjs.org/docs/react-component.html#setstate